### ISSUE TITLE
Not working with IE and Edge

### ISSUE THREAD

**Issue by Pilatch** [geocine/custom-elements-ts#46]
To reproduce:

    yarn install && yarn start site

Looks great in Chrome!

The code-example doesn't have any code to show inside of it in IE 10 and Edge.  I tested both using VirtualBox on my Mac, emulating Windows 10.

**Comment by Pilatch** [comment]
A problem with web components is that there isn't clear direction on how to make them work across a range of browsers, nor are there understandable manuals on how to build them.  I know there is abundant documentation for that purpose, but it's confusing and dense.  What a low-effort developer wants is a simple example.

Your library shows promise as an example that could be used for evangelizing the use of "vanilla" web components instead of resorting to Polymer, Vue, Riot or React or whatever.  That's why I'm here complaining about IE. I spent a good amount of time trying to find an existing, decent API and searching pages of NPM modules brought me to you.  My goal is to present a workflow to my coworkers and say, "We can do this. It's easy. It works everywhere we want it to, and eventually we won't need polyfills."

In my spare time I'm working toward that same purpose with this repository. https://github.com/Pilatch/simple-custom-element

In summary, the vanilla, cross-browser web component is vaporware.  That space deserves leadership, and I think your repository could provide that without much hassle.

**Comment by geocine** [comment]
This library was built out of need in an environment where we only use chrome. The goal is to add syntactic sugar against the existing web components api and that does not include polyfills. However, I will probably create a documentation how to use this with polyfills. Currently, the shadow dom stlye encapsulation polyfill deviates from how exactly we would use style encapsulation in the real shadow dom , https://github.com/webcomponents/shadycss . There is no straightforward way to do it. But the polyfill for custom element is quite straightforward https://github.com/webcomponents/custom-elements .
If you want to look into a much mature library which supports IE (specifically 11) you can check https://github.com/ionic-team/stencil.

**Comment by Pilatch** [comment]
Thanks! I'll check out stencil. It might be exactly what I want. :)

P.S. Regarding style encapsulation, that can be accomplished with BEM if you want styles to penetrate into custom elements. A good example is `font-family`, which you would expect to be the same as the rest of the page unless specified otherwise.  My requirements include a wide range of browser support, so we don't even consider shadow dom to be an option.
